File: ver_py/T11.py
# https://leetcode.cn/problems/recursive-mulitply-lcci/
class Solution:
    def multiply(self, A: int, B: int) -> int:
        if B == 0:
            return 0
        return A + self.multiply(A, B - 1)

if __name__ == '__main__':
    A, B = map(int, input().split())
    #s = Solution()
    print(Solution().multiply(A, B))

File: ver_cpp/T11.cpp
// https://leetcode.cn/problems/recursive-mulitply-lcci/
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    int multiply(int A, int B) {
        int Max = max(A, B), Min = min(A, B);
        if(Min==0) return 0;
        //Min * Max = (Min - 1 + 1) * Max = (Min - 1) *Max +Max 
        return multiply(Min - 1, Max) + Max;
    }
};